src/game.cpp

```diff
@@ -7968,9 +7984,89 @@ bool pet_menu(monster *z)
 
         return true;
     }
+    if ( inject == choice ) {
+        auto filter = [] (const item &it) {
+            return it.has_flag("MUTAGEN_CEPHALOPOD") ||
+                   it.has_flag("PURIFIER");
+        };
+        int pos = g->inv_for_filter( _("Injectable mutagen:"), filter );
+        if ( pos == INT_MIN ) {
+            add_msg( _("Never mind.") );
+            return true;
+        }
+
+        item *it = &g->u.i_at(pos);
+
+        if ( !( it->has_flag("MUTAGEN_CEPHALOPOD") ) &&
+             !( it->has_flag( "PURIFIER" ) ) ) {
+            add_msg(_("You can't inject that!"));
+            return true;
+        }
 
+        int mut_str = 0;
+        if ( !it->has_flag( "SERUM" ) ) {
+            mut_str = 1;
+        }
+        else {
+            mut_str = rng(1, 3);
+        }
+        if ( it->has_flag( "PURIFIER" ) ) {
+            mut_str *= -1;
+        }
+        if ( it->has_flag( "MUTAGEN_CEPHALOPOD" ) ||
+            it->has_flag( "PURIFIER" ) ) {
+
+            std::vector<mtype_id> dog_ceph_mut;
+            dog_ceph_mut.push_back(mtype_id("mon_dog"));
+            dog_ceph_mut.push_back(mtype_id("mon_beakhound"));
+            dog_ceph_mut.push_back(mtype_id("mon_beakhound2"));
+            dog_ceph_mut.push_back(mtype_id("mon_beakhound3"));
+            dog_ceph_mut.push_back(mtype_id("mon_beakhound4"));
+
+            int max_mut_lev = dog_ceph_mut.size() - 1;
+
+            for ( size_t i = 0; i < dog_ceph_mut.size(); i++ ) {
+                if (dog_ceph_mut[i] != z->type->id) {
+                    continue;
+                    }
+
+                int mut_lev = i + mut_str;
+
+                if (mut_lev > max_mut_lev) {
+                    mut_lev = max_mut_lev;
+                }
+                if (mut_lev < 0) {
+                    mut_lev = 0;
+                }
+
+                add_msg( _("You inject the %1$s into the %2$s."),
+                    it->type->nname(1).c_str(), pet_name.c_str() );
+                if (it->has_flag("SERUM")) {
+                    sounds::sound( z->pos(), 30, _("a howl!"));
+                    z->add_effect("wary", 1800);
+                    z->moves -= 500;
+                }
+
+                if ( z->type->id != dog_ceph_mut[mut_lev] && ( it->has_flag("SERUM") || one_in(4) ) ) {
+                    z->poly(dog_ceph_mut[mut_lev]);
+                    add_msg( _("The %s's form shifts right before your eyes!"),
+                        pet_name.c_str() );
+                } else {
+                    add_msg( _("Nothing seems to happen.") );
+                }
+                    
+                g->u.inv.use_charges( it->typeId(), 1 );
```

Comment:
Should be `g->u.use_charges`. Currently only charges in the main inventory of the character are used. The mutagen could be wielded or in a worn container.